WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressValidationViewController.swift

```diff
@@ -0,0 +1,309 @@
+import UIKit
+import Yosemite
+
+final class ShippingLabelAddressValidationViewController: UIViewController {
+
+    @IBOutlet private weak var tableView: UITableView!
+    @IBOutlet private weak var confirmButton: UIButton!
+
+    private let viewModel: ShippingLabelAddressValidationViewModel
+
+    /// Needed to scroll content to a visible area when the keyboard appears.
+    ///
+    private lazy var keyboardFrameObserver = KeyboardFrameObserver(onKeyboardFrameUpdate: { [weak self] frame in
+        self?.handleKeyboardFrameUpdate(keyboardFrame: frame)
+    })
+
+    /// Init
+    ///
+    init(addressVerification: ShippingLabelAddressVerification) {
```

Comment:
nit: it felt a bit odd to me seeing `ShippingLabelAddressVerification` passed to the address form VC, since `ShippingLabelAddressVerification` is the type sent to the Yosemite/Networking layer for the API validation call. How about we just pass `type` and `address` here, and let the view model assemble to `ShippingLabelAddressVerification` when it's making the validation call?
```suggestion
    init(type: ShipType, address: ShippingLabelAddress?) {
```

Reply:
It makes totally sense 👍